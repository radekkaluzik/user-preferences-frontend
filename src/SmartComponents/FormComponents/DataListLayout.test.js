import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DataListLayout from './DataListLayout';
import { Form, RendererContext } from '@data-driven-forms/react-form-renderer';

describe('DataListLayout checkbox tests', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Form onSubmit={() => undefined}>
        {(props) => (
          <RendererContext.Provider
            value={{
              formOptions: {
                renderForm: () => null,
                internalRegisterField: () => undefined,
              },
            }}
          >
            <DataListLayout
              label="test label"
              sections={[{ label: 'test', fields: [{ fields: [] }] }]}
              formOptions={{
                renderForm: () => 'test',
              }}
              clearedValue
              {...props}
            />
          </RendererContext.Provider>
        )}
      </Form>
    );
    expect(toJson(wrapper.find('DataListLayout'))).toMatchSnapshot();
  });
});
