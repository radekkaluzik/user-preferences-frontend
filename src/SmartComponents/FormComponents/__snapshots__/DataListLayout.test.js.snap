// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataListLayout checkbox tests should render correctly 1`] = `
<DataListLayout
  __versions={
    Object {
      "final-form": "4.20.4",
      "react-final-form": "6.5.7",
    }
  }
  clearedValue={true}
  dirty={false}
  dirtyFields={Object {}}
  dirtyFieldsSinceLastSubmit={Object {}}
  dirtySinceLastSubmit={false}
  errors={Object {}}
  form={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "restart": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
  formOptions={
    Object {
      "renderForm": [Function],
    }
  }
  handleSubmit={[Function]}
  hasSubmitErrors={false}
  hasValidationErrors={false}
  invalid={false}
  label="test label"
  modified={
    Object {
      "undefined": false,
    }
  }
  modifiedSinceLastSubmit={false}
  pristine={true}
  sections={
    Array [
      Object {
        "fields": Array [
          Object {
            "fields": Array [],
          },
        ],
        "label": "test",
      },
    ]
  }
  submitFailed={false}
  submitSucceeded={false}
  submitting={false}
  touched={
    Object {
      "undefined": false,
    }
  }
  valid={true}
  validating={false}
  values={Object {}}
  visited={
    Object {
      "undefined": false,
    }
  }
>
  <DataList
    __versions={
      Object {
        "final-form": "4.20.4",
        "react-final-form": "6.5.7",
      }
    }
    aria-label="test label"
    className=""
    clearedValue={true}
    dirty={false}
    dirtyFields={Object {}}
    dirtyFieldsSinceLastSubmit={Object {}}
    dirtySinceLastSubmit={false}
    errors={Object {}}
    form={
      Object {
        "batch": [Function],
        "blur": [Function],
        "change": [Function],
        "destroyOnUnregister": false,
        "focus": [Function],
        "getFieldState": [Function],
        "getRegisteredFields": [Function],
        "getState": [Function],
        "initialize": [Function],
        "isValidationPaused": [Function],
        "mutators": Object {},
        "pauseValidation": [Function],
        "registerField": [Function],
        "reset": [Function],
        "resetFieldState": [Function],
        "restart": [Function],
        "resumeValidation": [Function],
        "setConfig": [Function],
        "submit": [Function],
        "subscribe": [Function],
      }
    }
    formOptions={
      Object {
        "renderForm": [Function],
      }
    }
    gridBreakpoint="md"
    handleSubmit={[Function]}
    hasSubmitErrors={false}
    hasValidationErrors={false}
    invalid={false}
    isCompact={false}
    label="test label"
    modified={
      Object {
        "undefined": false,
      }
    }
    modifiedSinceLastSubmit={false}
    pristine={true}
    sections={
      Array [
        Object {
          "fields": Array [
            Object {
              "fields": Array [],
            },
          ],
          "label": "test",
        },
      ]
    }
    selectedDataListItemId=""
    submitFailed={false}
    submitSucceeded={false}
    submitting={false}
    touched={
      Object {
        "undefined": false,
      }
    }
    valid={true}
    validating={false}
    values={Object {}}
    visited={
      Object {
        "undefined": false,
      }
    }
    wrapModifier={null}
  >
    <ul
      __versions={
        Object {
          "final-form": "4.20.4",
          "react-final-form": "6.5.7",
        }
      }
      aria-label="test label"
      className="pf-c-data-list pf-m-grid-md"
      clearedValue={true}
      dirty={false}
      dirtyFields={Object {}}
      dirtyFieldsSinceLastSubmit={Object {}}
      dirtySinceLastSubmit={false}
      errors={Object {}}
      form={
        Object {
          "batch": [Function],
          "blur": [Function],
          "change": [Function],
          "destroyOnUnregister": false,
          "focus": [Function],
          "getFieldState": [Function],
          "getRegisteredFields": [Function],
          "getState": [Function],
          "initialize": [Function],
          "isValidationPaused": [Function],
          "mutators": Object {},
          "pauseValidation": [Function],
          "registerField": [Function],
          "reset": [Function],
          "resetFieldState": [Function],
          "restart": [Function],
          "resumeValidation": [Function],
          "setConfig": [Function],
          "submit": [Function],
          "subscribe": [Function],
        }
      }
      formOptions={
        Object {
          "renderForm": [Function],
        }
      }
      handleSubmit={[Function]}
      hasSubmitErrors={false}
      hasValidationErrors={false}
      invalid={false}
      label="test label"
      modified={
        Object {
          "undefined": false,
        }
      }
      modifiedSinceLastSubmit={false}
      pristine={true}
      sections={
        Array [
          Object {
            "fields": Array [
              Object {
                "fields": Array [],
              },
            ],
            "label": "test",
          },
        ]
      }
      submitFailed={false}
      submitSucceeded={false}
      submitting={false}
      touched={
        Object {
          "undefined": false,
        }
      }
      valid={true}
      validating={false}
      values={Object {}}
      visited={
        Object {
          "undefined": false,
        }
      }
    >
      <DataListItem
        aria-labelledby="simple-item1"
        className=""
        id=""
        isExpanded={false}
        key="0"
      >
        <li
          aria-labelledby="simple-item1"
          className="pf-c-data-list__item"
          id=""
        >
          <DataListRow
            fields={Array []}
            fieldsKey={0}
            formOptions={
              Object {
                "internalRegisterField": [Function],
                "renderForm": [Function],
              }
            }
            key=".$0-0"
            label="test"
            rowid="simple-item1"
          >
            <DataListItemRow>
              <div
                className="pf-c-data-list__item-row"
              >
                <DataListItemCells
                  dataListCells={
                    Array [
                      <DataListCell
                        className="pref-c-title pref-u-bold"
                        isFilled={false}
                      >
                        test
                      </DataListCell>,
                      <DataListCell
                        isFilled={true}
                      />,
                    ]
                  }
                  key=".0"
                  rowid=""
                >
                  <div
                    className="pf-c-data-list__item-content"
                  >
                    <DataListCell
                      className="pref-c-title pref-u-bold"
                      isFilled={false}
                      key="0-title"
                    >
                      <div
                        className="pf-c-data-list__cell pf-m-no-fill pref-c-title pref-u-bold"
                      >
                        test
                      </div>
                    </DataListCell>
                    <DataListCell
                      isFilled={true}
                      key="0-content"
                    >
                      <div
                        className="pf-c-data-list__cell"
                      />
                    </DataListCell>
                  </div>
                </DataListItemCells>
              </div>
            </DataListItemRow>
          </DataListRow>
        </li>
      </DataListItem>
    </ul>
  </DataList>
</DataListLayout>
`;
